### PR TITLE
Logger enhancement

### PR DESCRIPTION
**Pull request by MarrMika**

### CODE REVIEW

File: src/main/java/com/ita/if103java/ims/dao/impl/AbstractDao.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        logger.error(message, e);
+        return exception;
+    }
+
+    protected CRUDException crudException(RuntimeException e, String message) {
+        CRUDException exception = new CRUDException(e.toString());
+        logger.error(message, e);
+        return exception;
+    }
+
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/AbstractDao.java]: Говорили, балакали, сіли і заплакали

File: src/main/java/com/ita/if103java/ims/dao/impl/AbstractDao.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        EntityNotFoundException exception = new EntityNotFoundException(e.toString());
+        logger.error(message, e);
+        return exception;
+    }
+
+    protected CRUDException crudException(RuntimeException e, String message) {
+        CRUDException exception = new CRUDException(e.toString());
+        logger.error(message, e);
+        return exception;
+    }
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/AbstractDao.java]:
```
private static final Logger logger = LoggerFactory.getLogger(AbstractDao.class);

   protected EntityNotFoundException entityNotFoundException(EntityNotFoundException e, String message) {
        logger.error(message, e);
        return e;
    }

    protected CRUDException crudException(RuntimeException e, String message) {
        CRUDException exception = new CRUDException(e.toString());
        logger.error(message, e);
        return exception;
    }
```

File: src/main/java/com/ita/if103java/ims/dao/impl/AbstractDao.java (hunk 1 above)
Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/dao/impl/AbstractDao.java]: Сrying has not given freedom to anyone yet,
only who is the fighter that gains the world...

**Comment by ikrasWork** [comment]
Note to explain today:
```
@ControllerAdvice
public class GlobalControllerExceptionHandler {

    private static final Logger LOG = LoggerFactory.getLogger(GlobalControllerExceptionHandler.class);

    @ExceptionHandler(EntityNotFoundException.class)
    public ResponseEntity handleEntityNotFoundException(HttpServletRequest req, Exception ex) {
        LOG.warn(ex.getMessage(), ex);
        return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(new String[]{ex.getMessage()});
    }
}
```

**Pull request closed without merging**